core/src/main/java/tc/oc/pgm/listeners/PGMListener.java

```diff
@@ -258,6 +261,30 @@ public void handleItemPickup(final PlayerPickupItemEvent event) {
     if (nearestPlayer != event.getPlayer()) event.setCancelled(true);
   }
 
+  @EventHandler
+  public void lockFireTick(final MatchLoadEvent event) {
+    oldFireTick = Boolean.parseBoolean(event.getMatch().getWorld().getGameRuleValue(DO_FIRE_TICK));
+    event.getMatch().getWorld().setGameRuleValue(DO_FIRE_TICK, Boolean.toString(false));
+  }
+
+  // if no doFireTick <gamerule> is found, then use doFireTick setting saved in world
+  @EventHandler
+  public void unlockFireTick(final MatchStartEvent event) {
+    ImmutableMap<String, String> gameRules =
+        event.getMatch().getModule(GameRulesMatchModule.class).getGameRules();
```

Comment:
I'm not sure why `getGameRules` returning the whole map exists to start with, consider having a more specific getter that just tells you the value of the requested gamerule directly, will simplify this logic and logic elsewhere